healthcheck.go

```diff
@@ -53,6 +54,55 @@ type ClusterHealth struct {
 	Status string `json:"status"`
 }
 
+//indexcheck calls elasticsearch to check if the required indexes from the client exist
+func (cli *Client) indexcheck(ctx context.Context) (code int, err error) {
+
+	for _, index := range cli.indexes {
+
+		urlIndex := cli.url + "/" + index
+		logData := log.Data{"url": urlIndex, "index": index}
+
+		URL, err := url.Parse(urlIndex)
+		if err != nil {
+			log.Event(ctx, "failed to create url for elasticsearch indexcheck", log.ERROR, logData, log.Error(err))
+			return 500, err
+		}
+
+		path := URL.String()
+		logData["url"] = path
+
+		req, err := http.NewRequest("HEAD", path, nil)
+		if err != nil {
+			log.Event(ctx, "failed to create request for indexcheck call to elasticsearch", log.ERROR, logData, log.Error(err))
+			return 500, err
+		}
+
+		if cli.signRequests {
+			awsauth.Sign(req)
+		}
+
+		resp, err := cli.httpCli.Do(ctx, req)
+		if err != nil {
+			log.Event(ctx, "failed to call elasticsearch", log.ERROR, logData, log.Error(err))
+			return 500, err
+		}
+		defer resp.Body.Close()
+		logData["http_code for"] = resp.StatusCode
+
+		switch resp.StatusCode {
+		case 200:
+			continue
+		case 404:
+			log.Event(ctx, "index does not exist", logData, log.ERROR, log.Error(ErrorIndexDoesNotExist))
+			return resp.StatusCode, ErrorIndexDoesNotExist
```

Comment:
Just an idea: would it be worth completing the loop, and keeping an array of indexes that are not found, so that if more than one index is missing, we will be able to provide a full list of missing indexes in the checker message?
Maybe in a future task?

Reply:
@janderson2 with this suggestion shall I do this task as part of my ticket?